spec/api-net-spec.js

```diff
@@ -1375,36 +1376,90 @@ describe('net module', () => {
             handleUnexpectedURL(request, response)
         }
       })
+
       const urlRequest = net.request({
         method: 'GET',
         url: `${server.url}${requestUrl}`
       })
+
       urlRequest.on('response', (response) => {
         const statusCode = response.statusCode
         assert(typeof statusCode === 'number')
         assert.strictEqual(statusCode, 200)
+
         const statusMessage = response.statusMessage
         assert(typeof statusMessage === 'string')
         assert.strictEqual(statusMessage, 'OK')
+
         const headers = response.headers
         assert(typeof headers === 'object')
-        assert.deepStrictEqual(headers[customHeaderName.toLowerCase()],
-          [customHeaderValue])
+        assert.deepStrictEqual(headers[customHeaderName.toLowerCase()], customHeaderValue)
+
         const httpVersion = response.httpVersion
         assert(typeof httpVersion === 'string')
         assert(httpVersion.length > 0)
+
         const httpVersionMajor = response.httpVersionMajor
         assert(typeof httpVersionMajor === 'number')
         assert(httpVersionMajor >= 1)
+
         const httpVersionMinor = response.httpVersionMinor
         assert(typeof httpVersionMinor === 'number')
         assert(httpVersionMinor >= 0)
+
         response.pause()
-        response.on('data', (chunk) => {
-        })
-        response.on('end', () => {
-          done()
-        })
+        response.on('data', (chunk) => {})
+        response.on('end', () => { done() })
+        response.resume()
+      })
+      urlRequest.end()
+    })
+
+    it('should discard duplicate headers', (done) => {
+      const requestUrl = '/duplicateRequestUrl'
+      const includedHeader = 'max-forwards'
+      const discardableHeader = 'Max-Forwards'
+
+      const includedHeaderValue = 'max-fwds-val'
+      const discardableHeaderValue = 'max-fwds-val-two'
+
+      server.on('request', (request, response) => {
+        switch (request.url) {
+          case requestUrl:
+            response.statusCode = 200
+            response.statusMessage = 'OK'
+            response.setHeader(includedHeader, includedHeaderValue)
+            response.setHeader(discardableHeader, discardableHeaderValue)
+            response.end()
+            break
+          default:
+            handleUnexpectedURL(request, response)
+        }
+      })
+
+      const urlRequest = net.request({
+        method: 'GET',
+        url: `${server.url}${requestUrl}`
+      })
+
+      urlRequest.on('response', (response) => {
+        const statusCode = response.statusCode
+        assert(typeof statusCode === 'number')
+        assert.strictEqual(statusCode, 200)
```

Comment:
```suggestion
        expect(statusCode).to.be(200)
```

No need to check that `typeof x === 'number'` if you're going to check that it's `strictEqual` to a number.

Reply:
i don't think we have expect in this file so i was matching previous style but i can add it if that seems like a better way forward :)

Reply:
oh weird, yeah we shouldn't be adding new code in the old style IMO.